Assets/Scripts/Shop/ShopUIController.cs

```diff
@@ -0,0 +1,112 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class ShopUIController : MonoBehaviour
+{
+    private Transform container;
+    private Transform shopItemTemplate;
+    private List<Transform> shopItems = new List<Transform>();
+    private int selectedItemIndex = 0;
+
+    private void Awake()
+    {
+        container = transform.Find("Container");
+        shopItemTemplate = container.Find("ShopItemTemplate");
+        shopItemTemplate.gameObject.SetActive(false);
+    }
+
+    private void Start()
+    {
+        CreateItemButton("Damage Item", Item.ItemType.Damage_1);
+        CreateItemButton("Damage Item 2", Item.ItemType.Damage_2);
+
+        UpdateSelection();
+        Hide();
+    }
+
+    private void Update()
+    {
+        if (!gameObject.activeSelf) return;
+
+        if (Input.GetKeyDown(KeyCode.DownArrow))
```

Comment:
Shouldn't the user use their mouse buttons to interact with the shop? I believe on-screen 2D Buttons would make your job a lot easier